scripts/uberenv/packages/hip/package.py

```diff
@@ -0,0 +1,57 @@
+from spack import *
+
+
+class Hip(CMakePackage):
+    """HIP is a C++ Runtime API and Kernel Language that allows developers to
+       create portable applications for AMD and NVIDIA GPUs from
+       single source code."""
+
+    homepage = "https://github.com/ROCm-Developer-Tools/HIP"
+    url      = "https://github.com/ROCm-Developer-Tools/HIP/archive/rocm-3.5.0.tar.gz"
+
+    maintainers = ['srekolam', 'arjun-raj-kuppala']
+
+    version('3.7.0', sha256='757b392c3beb29beea27640832fbad86681dbd585284c19a4c2053891673babd')
+    version('3.5.0', sha256='ae8384362986b392288181bcfbe5e3a0ec91af4320c189bd83c844ed384161b3')
+
+    depends_on('cmake@3:', type='build')
+    depends_on('perl@5.10:', type=('build', 'run'))
+    depends_on('mesa~llvm@18.3:')
+
+    for ver in ['3.5.0', '3.7.0']:
+        depends_on('rocclr@' + ver,  type='build', when='@' + ver)
+        depends_on('hsakmt-roct@' + ver, type='build', when='@' + ver)
+        depends_on('hsa-rocr-dev@' + ver, type='link', when='@' + ver)
+        depends_on('comgr@' + ver, type='build', when='@' + ver)
+        depends_on('llvm-amdgpu@' + ver, type='build', when='@' + ver)
+        depends_on('rocm-device-libs@' + ver, type='build', when='@' + ver)
+        depends_on('rocminfo@' + ver, type='build', when='@' + ver)
+
+    # Notice: most likely this will only be a hard dependency on 3.7.0
+    depends_on('numactl', when='@3.7.0')
+
+    def setup_dependent_package(self, module, dependent_spec):
+        self.spec.hipcc = join_path(self.prefix.bin, 'hipcc')
+
+    @run_before('install')
+    def filter_sbang(self):
+        perl = self.spec['perl'].command
+        kwargs = {'ignore_absent': False, 'backup': False, 'string': False}
+
+        with working_dir('bin'):
+            match = '^#!/usr/bin/perl'
+            substitute = "#!{perl}".format(perl=perl)
+            files = [
+                'hipify-perl', 'hipcc', 'extractkernel',
+                'hipconfig', 'hipify-cmakefile'
+            ]
+            filter_file(match, substitute, *files, **kwargs)
+
+    def cmake_args(self):
+        args = [
+            '-DHIP_COMPILER=clang',
+            '-DHIP_PLATFORM=rocclr',
+            '-DHSA_PATH={0}'.format(self.spec['hsa-rocr-dev'].prefix),
+            '-DLIBROCclr_STATIC_DIR={0}/lib'.format(self.spec['rocclr'].prefix)
+        ]
+        return args
```

Comment:
The hip package already exists in Spack. Why was this necessary?
Could you in particular delimit the changes with comments like `# BEGIN/END Umpire changes`?

Reply:
The package in Spack is completely broken for external installs of HIP.